backend/controllers/auth_controller.go

```diff
@@ -0,0 +1,92 @@
+package controllers
+
+import (
+	"net/http"
+
+	"github.com/ZPI-2024-25/KubernetesAccessManager/auth"
+	"github.com/ZPI-2024-25/KubernetesAccessManager/models"
+	"github.com/golang-jwt/jwt/v4"
+)
+
+func CheckLoginStatus(w http.ResponseWriter, r *http.Request)  {
+	token, err := auth.GetJWTTokenFromHeader(r)
+	isValid, claims := auth.IsTokenValid(token)
+
+	if err != nil || !isValid {
+		writeJSONResponse(w, http.StatusUnauthorized, models.ModelError{
+			Message: "Unauthorized",
+			Code: http.StatusUnauthorized,
+		})
+		return
+	}
+	rolemap, err := auth.GetRoleMapInstance()
+	if err != nil {
+		writeJSONResponse(w, http.StatusInternalServerError, &models.ModelError{
+			Message: "Internal Server Error",
```

Comment:
Why don't you use error message, from err var, that tells much more precisely what problem occurred?

Reply:
```suggestion
			Message: fmt.Sprintf("Failed to get client: %s", err),
```